test/ao/sparsity/test_pruner.py

```diff
@@ -182,136 +213,201 @@ def test_prepare_linear_bias(self):
             assert type(module.parametrizations.weight[0]) == PruningParametrization
         assert model(x).shape == (128, 16)
 
-    def test_convert_linear(self):
+    def test_prepare_linear_bias(self):
+        self._test_prepare_linear_bias_on_device(torch.device('cpu'))
+        if torch.cuda.is_available():
+            self._test_prepare_linear_bias_on_device(torch.device('cuda'))
+
+    def _test_convert_linear_on_device(self, device):
+        print(f'Testing on {device.type}')
         model = Linear()
+        model = model.to(device)
         x = torch.ones(128, 16)
         pruner = SimplePruner(model, None, None)
         pruner.prepare()
         pruner.convert()
         for g in pruner.module_groups:
             module = g['module']
+            assert module.weight.device == device
             assert not hasattr(module, "parametrizations")
             assert not hasattr(module, 'mask')
         assert model(x).shape == (128, 16)
 
-    def test_convert_linear_bias(self):
+    def test_convert_linear(self):
+        self._test_convert_linear_on_device(torch.device('cpu'))
+        if torch.cuda.is_available():
+            self._test_convert_linear_on_device(torch.device('cuda'))
+
+    def _test_convert_linear_bias_on_device(self, device):
```

Comment:
it looks like there is still a lot of duplications in these code, is it possible to simplify further?